framework/src/components/storage/entities/account.js

```diff
@@ -677,6 +720,88 @@ class Account extends BaseEntity {
 		);
 	}
 
+	/**
+	 * Update dependent records for the account
+	 * Delete dependent record for the account
+	 *
+	 * @param {string} dependencyName - Name of the dependent table
+	 * @param {string} address - Address of the account
+	 * @param {string} dependentPublicKey - Dependent public id
+	 * @param {Object} [tx] - Transaction object
+	 * @return {*}
+	 */
+	updateDependentRecords(dependencyName, address, dependentPublicKeys, tx) {
+		assert(
+			Object.keys(dependentFieldsTableMap).includes(dependencyName),
+			`Invalid dependency name "${dependencyName}" provided.`
+		);
+
+		const sqlForDelete = this.SQLs.deleteDependentRecords;
+
+		const tableName = dependentFieldsTableMap[dependencyName];
+		const sqlForInsert = this.SQLs.createDependentRecords;
+
+		return this.adapter
+			.execute(
+				`SELECT "dependentId" FROM ${tableName} WHERE "accountId" = $1`,
+				[address]
+			)
+			.then(dependentRecordsForAddress => {
+				const oldDependentPublicKeys = dependentRecordsForAddress.map(
+					dependentRecord => dependentRecord.dependentId
+				);
+				const toRemove = oldDependentPublicKeys.filter(
+					aPK => !dependentPublicKeys.includes(aPK)
+				);
+				const toInsert = dependentPublicKeys.filter(
```

Comment:
Maybe we can articulate the variable names in this function better, and organize `const` variable declerations.

Reply:
Done